**Step 1: Use a SELECT query to select all records from the customer table. Limit the displayed results to 10 records.**

In [3]:
SELECT *
FROM dsci_504.customers
LIMIT 10;

(10 row(s) affected)

Total execution time: 00:00:00.005

cus_id cus_num cus_last_name cus_first_name cus_add_num cus_address cus_city cus_state cus_zip cus_phone cus_join_date cus_app_cd cus_app_num tot_ord_qty tot_ord_value cus_app_lvl 471 Smi63500 Smith Taylor 67500 Beckley Hitt 20 34262 6449498245 2016-08-22 1 OPC87242708 48 31106 Low 627 Pre95680 Presley Erin 33079 Colonial Panda Pike Howe 5 31456 7227621838 2018-11-26 1 OPC81648129 64 21532 Low 1255 Coo931 Cooper Carl 54499 Blue Stead North Brewster 16 97377 7462214127 2016-03-27 1 OPC81381401 97 16208 Low 2283 Tan94186 Tankersly Joshua 27541 Bean Creek Montello 25 76506 95964736 2005-05-06 4 OPC84608351 8 14859 Low 2241 Jam71966 Jamieson Noel 81291 Quaking Embers Quay Cowan 10 82205 7971817866 2012-11-24 2 OPC79124424 52 24171 Low 2617 NULL Mathiasen Ulrich 155 Foxrun Loganville 9 30052 2948717959 2020-01-27 18 OPC85649476 45 27099.12 High 21 Jon39709 Jones Piper 75162 Rustic Grove Orchard Laffertys Trailer Court 28 8544 1992380878 2019-02-18 1 OPC87398388 23 32288 Low 22 Rid36041 Ridgley Brian 25138 Crystal Cove North Star 23 70033 5593513806 2011-05-18 7 OPC80243377 95 12960 Moderate 23 Sei48735 Seiler Mike 64092 Honey Creek Autoroute Norton Shores 1 78116 476557784 2009-07-15 2 OPC79478730 86 343 Low 24 Smi73882 Smith Erin 29529 Hazy Road Holiday Park 19 36253 4748917378 2016-02-10 7 OPC85629135 24 18141 Moderate

**Q: Of the first ten customers returned, what is the Customer ID  and City of the 5th customer?**

```
cus_id for 5th customer = 2241

```
```
cus_city for 5th customer = Cowan

```

In [9]:
select customers.cus_id,customers.cus_city
from dsci_504.customers
LIMIT 1 OFFSET 4;

(1 row(s) affected)

Total execution time: 00:00:00.003

cus_id cus_city 2241 Cowan

**Step 2: Use a SELECT query to select the Customer ID, Last Name, State, and Customer Appreciation Code of all customers who joined _from_ September 11, 2001 _through_ October 1, 2001**

In [10]:
SELECT
    cus_id,
    cus_last_name,
    cus_state,
    cus_app_cd
FROM
    dsci_504.customers
WHERE
   cus_join_date >= '2001-09-11' AND cus_join_date <= '2001-10-01';


(8 row(s) affected)

Total execution time: 00:00:00.006

cus_id cus_last_name cus_state cus_app_cd 356 Delacruz 2 6 464 Sherman 3 1 495 Delacruz 18 1 1587 Applewood 17 4 1907 Acosta 29 3 1936 Manchester 11 6 2141 Brooks 19 4 2194 Moore 28 4

**Q: How many customers joined in teh period from September 11, 2001 throught October 1, 2001?**

```
A: eight customers joined (8)

```

**Part 3: Looking at the same query from Part 2, write a SELECT query that returns all information for the two customers with the last name Delacruz**

In [11]:
SELECT *
FROM dsci_504.customers
where cus_join_date >= '2001-09-11' AND cus_join_date <= '2001-10-01'
AND cus_last_name = 'Delacruz';

(2 row(s) affected)

Total execution time: 00:00:00.005

cus_id cus_num cus_last_name cus_first_name cus_add_num cus_address cus_city cus_state cus_zip cus_phone cus_join_date cus_app_cd cus_app_num tot_ord_qty tot_ord_value cus_app_lvl 356 Del73426 Delacruz Margo 63740 Crystal Spring Corners Mcarthur Estates 2 89443 3875686732 2001-09-25 6 OPC81506974 70 34773 Moderate 495 Del44118 Delacruz Brandon 53556 Apache Wolfs Mill 18 18151 7593410556 2001-09-23 1 OPC84687770 85 2396 Low

**Q: What are three unique features about these two customers?**

**Customer ID: Each customer has a unique Customer ID.**

**Customer 1: 356**

**Customer 2: 495**

**Join Date: The customers have different join dates.**

**Customer 1: 2001/09/25**

**Customer 2: 2001/09/23**

**State: They have different states associated with them.**

**Customer 1: State 2**

**Customer 2: State 18**

**Part 4: Write a query to return all order data for the two Delacruz customers above. Be sure to include minimal person information to assist in identifying which person placed which order. You may need to use a simple join to get both the customer and order data. A template for that is below. Use the following data points in your query:**

- **Customer ID**
- **Customer Last Name**
- **Customer First Name**
- **Total Order Quantity**
- **Total Order Value**
- **All Order Data**

```
SELECT [table1.column], [table2.column]..
FROM [table1], [table2]
WHERE [conditions]

```

In [13]:
SELECT
    Customers.cus_id,
    Customers.cus_last_name,
    Customers.cus_first_name,
    Customers.tot_ord_qty,
    Customers.tot_ord_value,
    orders.*
FROM
    dsci_504.Customers,   
    dsci_504.orders
WHERE
    Customers.cus_id = Orders.cus_id   
AND
    Customers.cus_last_name = 'Delacruz'
AND 
    cus_join_date >= '2001-09-11' AND cus_join_date <= '2001-10-01'  
   



(2 row(s) affected)

Total execution time: 00:00:00.007

cus_id cus_last_name cus_first_name tot_ord_qty tot_ord_value ord_id ord_date ord_tax_loc order_tot ord_ship_add ord_ship_date ord_track_num prod_id warehouse_id cus_id build_id 356 Delacruz Margo 70 34773 2614702 2002-08-21 2 5200 false 2002-08-25 PAC7774446-3 42 3 356 37 495 Delacruz Brandon 85 2396 2051970 2005-07-28 18 5631.35 false 2005-07-30 PAC2382180-2 48 1 495 50

**Q: What is one interesting fact revealed in addition** **the new data?**

**Customer Margo Delacruz (Customer ID: 356) has  a tot\_ord\_qty of  70, tot\_ord\_value of 34,773, and ord\_id 2614702**

**Customer Brandon Delacruz (Customer ID: 495)**  **has  a tot\_ord\_qty of** **85****, tot\_ord\_value** **of 2,396 and ord\_id 2051970**

**This indicates that these two customers have distinct preferences.**

**Q: Explain how the addition of additional data aided in your assessment.**

```
The additional data allowed a more comprehensive assessment of these customers' profiles and behaviors. It is valuable for businesses aiming to personalize customer interactions and enhance customer satisfaction and retention.

```

**Part 5: Some of the most unique features mountain bikers seek out in their bikes are wheel sets, shocks, forks, and dropper posts. These items make or break a top- notch mountain bike. You know that part of the identification of whether or not OPC should break into the luxury MTB market is that they can build and market competitive bikes. Use a SELECT query to determine how many unique type of components there are in the components table. Specifically query for forks and dropper posts. Order your results by component category and component name in alphabetical order.**

In [16]:
SELECT DISTINCT
    components.comp_name,
    Components.comp_cat
FROM
    dsci_504.components
WHERE
    Components.comp_cat IN ('fork', 'dropper')
ORDER BY
    components.comp_name ASC,
    Components.comp_cat ASC;

(85 row(s) affected)

Total execution time: 00:00:00.004

comp_name comp_cat Bike Yoke Devine Stealth dropper Bike Yoke Revive dropper BrandX Ascend dropper Cannondale 2 dropper Cannondale DownLow dropper Cube Dropper dropper DropWH18 dropper DVO Onyx D1 160 fork DVO Onyx D2 170 Air fork Fox 32 Factory 100 fork Fox 32 Float Perf 100 Air fork Fox 34 Float Factory Grip 2 120 fork Fox 34 Float Factory Grip 2 140 fork Fox 34 Float Factory Grip 2 140 Air fork Fox 34 Float Perf 120 fork Fox 34 Float Perf 140 fork Fox 34 Float Perf 140 Air fork Fox 34 Perf 130 fork Fox 34 Perf 130 Air fork Fox 34 Rhythm 120 fork Fox 36 Factory 150 Air fork Fox 36 Factory 160 Air fork Fox 36 Float Factory 160 Air fork Fox 36 Float Factory 170 Air fork Fox 36 Float Perf 140 Air fork Fox 36 Float Perf 150 fork Fox 36 Float Perf 160 Air fork Fox 36 Perf 160 Air fork Fox 36 Perf 1650 Air fork Fox 36 Perf 170 Air fork Fox 36 Perf Elite 140 Air fork Fox 36 Perf Elite 150 Air fork Fox 36 Perf Elite 160 Air fork Fox 36 Perf Elite 180 Air fork Fox 36 Rhythm 160 Air fork Fox 36 Rhythm Fit 3 150 fork Fox 38 Factory Grip 2 180 Air fork Fox 38 Factory Grip2 170 Air fork Fox 38 Float Perf 170 Air fork Fox 38 Grip 2 170 Air fork Fox Transfer dropper Fox Transfer Factory dropper FusionX Manic dropper Giant Contact Switch dropper GT Dropkick dropper Ibis dropper KS Lev dropper KS Lev Integra dropper KS Lev SI dropper KS Rage-i dropper Lefty Ocho 100 Air fork Marzocchi Bomber Z2 130 Air fork Marzocchi Bomber Z2 Rail 120 Air fork Niner RDO dropper OC2 dropper OC2 Carbon dropper Ohlins RXF36 M.2 170 Air fork OneUp V2 dropper OnOff Pija dropper Recon Dropper dropper Recon Elite Dropper dropper Ride Alpha dropper RockShox 35 Gold RL 140 Air fork RockShox 35 Gold RL 150 Air fork RockShox 35 Gold RL 160 Air fork RockShox Lyric Sel 150 Air fork RockShox Lyric Ult 150 Air fork RockShox Lyric Ult RC2 160 fork RockShox Lyric Ult RC2 160 Air fork RockShox Lyric Ult RC2 170 fork RockShox Pike Sel RC 140 fork RockShox Pike Sel+ 140 Air fork RockShox Pike Ult 150 Air fork RockShox Pike Ult RC 130 Air fork RockShox Recon RL 120 fork RockShox Recon RL 120 Air fork RockShox Recon RL 130 fork RockShox Recon RL 130 Air fork RockShox Reverb Steakth dropper RockShox YariRC 160 fork RockShox YarRC 180 Air fork RockShox ZEB Ult 170 Air fork SDG Tellis dropper TransX dropper X-Fusion Manix dropper

**Q: How many unique combination of fork and dropper post were there in the data?**

```
A: 85

```

**Part 6: Write a new query to identify all builds that the fork in row 60 is included on in the data. Include the accompanying build data in your query. Use a conditional query and search for the component name in the query.**

In [21]:
SELECT builds.*,components.comp_name
FROM dsci_504.builds,dsci_504.components
where builds.build_id = '60'
and components.comp_name = 'Sram Code RS';




(1 row(s) affected)

Total execution time: 00:00:00

build_id comp_hbar comp_brake comp_wheels comp_fork comp_shock comp_shifter comp_cassette comp_derail comp_dropper comp_crank comp_chain_ring build_name comp_name 60 40 63 86 141 193 275 275 275 250 275 275 quick strike Sram Code RS

**Q: What happened in the query?**

```
A: the query gave me the build name and all information when the build id = 60 and component name = Sram Code RS

```

**Part 7: Use the simple join concept from Part 4 to complete the query by joining the componet and build tables. Return all Build columns in your query, but not the component data.**

In [24]:
SELECT Builds.*
FROM dsci_504.builds, dsci_504.components
where builds.comp_fork = '123'
and  Components.comp_id ='123'



(4 row(s) affected)

Total execution time: 00:00:00.004

build_id comp_hbar comp_brake comp_wheels comp_fork comp_shock comp_shifter comp_cassette comp_derail comp_dropper comp_crank comp_chain_ring build_name 71 43 67 91 123 184 61 219 273 260 273 273 precision 42 30 67 88 123 184 61 219 273 260 273 273 electric 45 26 67 72 123 192 273 274 229 259 273 273 megalodon 57 6 56 85 123 188 279 223 279 237 269 279 strikeforce

**Q: How many builds is the 'Fox 38 Factory Grip2 170 Air' included in?**

```
A: four (4)

```

**Part 8: Write a query to find the build names of the four builds from Part 7 above.**

In [25]:
SELECT Builds.build_name,components.comp_name
FROM dsci_504.builds, dsci_504.components
where builds.comp_fork = '123'
and  Components.comp_id ='123'


(4 row(s) affected)

Total execution time: 00:00:00.003

build_name comp_name precision Fox 38 Factory Grip2 170 Air electric Fox 38 Factory Grip2 170 Air megalodon Fox 38 Factory Grip2 170 Air strikeforce Fox 38 Factory Grip2 170 Air

**Part 9: Identify the most expensive product of 2019. Since we haven't learned aggregate functions yet, please do not use the max() function. Use conditional statements to return a single row with the product with the highest order value**

In [70]:
SELECT products.prod_name, products.prod_price, orders.order_tot
FROM dsci_504.orders,dsci_504.products 
WHERE orders.prod_id = products.prod_id
AND orders.ord_date >= '2019-01-01' AND orders.ord_date <= '2019-12-31'
ORDER BY products.prod_price   DESC
limit 1;

(1 row(s) affected)

Total execution time: 00:00:00.004

prod_name prod_price order_tot Rail 7399 7813.34

**Q: What is the product name of the product and the price of the mountain bike with the highest prive ordered in 2019?**

```
A: rail = 7399

```

**Part 10: Execute three additional queries on tables you haven't used yet. These can be at your discretion.**

In [72]:
-- select order id , date and total where order ship = false
SELECT orders.ord_id, orders.ord_date, orders.order_tot
FROM dsci_504.orders
WHERE orders.ord_ship_add = 'false';


(1326 row(s) affected)

Total execution time: 00:00:00.018

ord_id ord_date order_tot 4825298 2017-03-28 5275 7009183 2010-10-14 4503.2 5186311 2001-05-12 8073.44 4274782 2002-03-19 3572.2 2579402 2012-11-10 4158.96 6921547 2012-02-26 5006.53 6707258 2008-07-03 5095.44 2553970 2002-05-22 3640 4160529 2007-10-08 3905.39 5140837 2003-09-29 3999 2715724 2013-09-05 5856.44 4066723 2019-03-28 4399 7151327 2013-11-27 6024.08 1912588 2010-12-18 3999 1449568 2019-02-25 4449.9 5155394 2011-04-25 5856.44 4595626 2003-06-19 2066.99 1895617 2003-11-07 8320 3417447 2004-10-20 4685.19 4533036 2013-10-18 1951.75 5440045 2010-04-12 5698.55 3846551 2014-03-05 4640.95 4679322 2000-12-26 4924.75 5835237 2004-03-27 3948.38 3333624 2016-01-13 6252.94 4305349 2006-10-03 6794.27 4469762 2008-01-09 4706.93 6072149 2013-05-07 5736.5 2016400 2015-04-10 4574.96 3608797 2008-11-13 2086.49 4044365 2010-09-08 4982 2900047 2008-07-08 4200 2588796 2001-06-08 4600.99 3485081 2014-11-15 6058.1 2358239 2018-06-10 3199 7112703 2009-11-18 6146.94 4185159 2001-03-24 3399 1526982 2002-05-01 6810.68 3357050 2002-03-19 3807 6771690 2010-12-07 4609 1911288 2014-12-16 4450.94 1930308 2004-09-19 4158.96 4218977 2015-04-23 3798 5186226 2012-02-13 5752.31 2208177 2008-11-16 4664 5788991 2009-05-02 4880.2 5105818 2000-10-12 3672.44 4157138 2004-07-08 5302.96 3031547 2013-06-08 3534.96 6960628 2018-10-23 4797.81 4054195 2002-01-25 5510.94 3937927 2017-02-08 4523.95 3307281 2015-08-24 4416.14 6585838 2010-03-09 5406.96 3220422 2004-01-04 4018.48 6711968 2019-05-02 3999.99 5065896 2009-06-06 4627.9 3624763 2006-04-03 5654.21 7187392 2008-03-02 2797.34 3364980 2005-09-07 2330.94 3178673 2006-05-20 4192.95 4521021 2007-02-26 3430.93 1539389 2017-07-21 3689.85 6293114 2011-12-10 4519.88 7197660 2017-06-24 1932.33 5305378 2000-06-01 6395 2356703 2004-09-19 4543.58 5242761 2010-06-06 4524 4080696 2017-08-20 2199 3570845 2011-04-06 6220.5 4389481 2007-06-14 2350.18 6953949 2013-02-15 6693.46 2748922 2014-11-21 3999 1409838 2005-07-25 4176.96 4882154 2007-12-27 4222.94 1499647 2016-09-27 7399 5775992 2007-07-20 5659.99 4711571 2001-05-21 5591.03 5687001 2011-09-16 5375.99 6077808 2010-03-25 4218.95 6241665 2009-01-22 3602.94 6047294 2014-06-16 8200.49 3236261 2001-10-26 4701.43 5868564 2017-10-03 3399 5255967 2006-04-05 5800 6894826 2015-02-13 4273.93 3560653 2011-01-11 4158.96 3779453 2004-06-30 7757.85 7311476 2012-05-20 3001.93 2177932 2016-01-19 3745 4205221 2006-02-22 3710 3193964 2007-06-02 5609.99 6464145 2015-04-07 7842.94 5236088 2015-06-03 4158.96 3681241 2004-05-08 4611 1929235 2001-08-05 6842.65 1694460 2019-07-11 5995.79 6755650 2018-02-01 4761.39 2629583 2008-08-23 5499 1458147 2006-07-25 2341.94 5651788 2000-07-26 4400 6556364 2017-09-23 4557.99 3272347 2005-02-11 5830 2845076 2004-08-27 3390.94 3293788 2014-08-31 3199 2679922 2011-01-08 4409.79 3872219 2003-03-26 4228.94 5417203 2015-10-14 5967.51 6976313 2006-10-07 4040.24 3353836 2006-06-17 4260.13 2140430 2012-08-11 3401.62 1724480 2018-02-17 4350 5385043 2016-11-04 3195.75 7372789 2000-11-24 4274.2 2542761 2017-08-20 3707.25 2796751 2016-10-13 4300.32 4706940 2003-09-14 6177 7083064 2008-02-18 6134.96 4366976 2018-06-18 5613.65 2069011 2000-06-01 3673.99 5968744 2009-03-29 4433.05 3984108 2017-04-17 4934.8 4460714 2017-08-14 5634.44 4206987 2016-05-04 4238.94 1332784 2015-04-27 4982.82 7293392 2005-07-01 3189.54 2678429 2011-06-30 2752.48 3758613 2004-08-30 5946.59 5780157 2015-12-11 4678.96 6419973 2017-11-29 4685.19 5046694 2010-07-13 3337.44 1450563 2003-01-04 2639 2357460 2007-03-14 1949.99 4433548 2015-05-17 4797.81 1229905 2004-07-22 8065.74 4464272 2001-10-07 4258.94 4606107 2014-05-21 4589.25 2804411 2015-10-15 7842.94 4332112 2009-10-18 4222.94 1588017 2005-07-01 5484.95 3670818 2014-12-18 1961 3094136 2011-11-10 5575.93 2640087 2014-03-25 6238.19 4860434 2016-04-16 5600.41 5207861 2016-09-30 3534.96 2581918 2004-04-17 6590.49 5635476 2019-12-11 3837.48 5116839 2000-03-14 3944.65 2993361 2005-10-18 2084.05 6860822 2002-02-16 4260.13 4988395 2013-11-02 6326.68 121

In [74]:
-- select all data from taxes table when the rate more than 2
select *
FROM dsci_504.taxes
where taxes.tax_rate > 2

(10 row(s) affected)

Total execution time: 00:00:00.002

tax_id tax_loc tax_rate 4 CA 8.66 9 GA 7.31 18 NC 6.97 20 OH 7.17 23 RI 7.00 25 TN 8.52 26 UT 7.18 31 NY 8.52 22 PA 6.34 13 KS 8.52

In [75]:
-- select all data from customerhouseorders limit 5
SELECT *
FROM dsci_504.customerwarehouseorders
LIMIT 5;

(5 row(s) affected)

Total execution time: 00:00:00.004

cus_id warehouse_id ord_id 27 2 5920416 28 3 4825298 29 1 7009183 30 1 5186311 31 3 4274782

**Q: Write a few paragraphs about what you learned this week in the course. What exciting things did you discover in this assignment with simple queries?**

```
A: This assignment has provided a practical introduction to SQL querying, highlighting the importance of retrieving, filtering, and joining data to extract meaningful insights from a relational database. SQL is a powerful tool for data manipulation and retrieval, and understanding its fundamentals is crucial for anyone working with databases and data analysis.

```